textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/planstore/PlanStore.java

```diff
@@ -194,6 +235,40 @@ public void updatePlan(String planName, String description, LogicalPlan plan) th
     }
 
     /**
+     * Updates plan object of a plan with the given plan name.
+     *
+     * @param planName, the name of the plan.
+     * @param logicalPlanJson, the new plan object.
+     * @throws TextDBException
+     */
+    public void updatePlanJson(String planName, LogicalPlan logicalPlanJson) throws TextDBException {
```

Comment:
Yes it is going to be a `String`. I am so sorry, this was before our discussion and is still a WIP.   Please have a look and review it tomorrow.